packages/contracts/src/ERC20KPIToken.sol

```diff
@@ -86,15 +86,16 @@ contract ERC20KPIToken is ERC20Upgradeable, IERC20KPIToken, ReentrancyGuardUpgra
     error NothingToRecover();
     error NotEnoughValue();
 
-    event Initialize();
     event CollectProtocolFee(address indexed token, uint256 amount, address indexed receiver);
-    event OwnershipTransferred(address indexed oldOwner, address indexed newOwner);
-    event Finalize(address indexed oracle, uint256 result);
     event RecoverERC20(address indexed token, uint256 amount, address indexed receiver);
     event Redeem(address indexed account, uint256 burned);
     event RegisterRedemption(address indexed account, uint256 burned);
     event RedeemCollateral(address indexed account, address indexed receiver, address token, uint256 amount);
 
+    constructor() {
+        _disableInitializers();
```

Comment:
What is this for?

Reply:
Check this out, in particular the warning boxes right below: https://docs.openzeppelin.com/contracts/4.x/api/proxy#Initializable